testing/tools/integration/external.py

```diff
@@ -178,7 +182,9 @@ def save_logs_to_file(base_dir, log_stream=None, persistent_data={}):
         for sk in sink_data:
             sink_log_name = 'sink_{address}_{time}.error.dat'.format(
                 address=sk.address.replace(':', '.'),
-                time=sk.start_time.strftime('%Y%m%d_%H%M%S.%f'))
+                time=(sk.start_time.strftime('%Y%m%d_%H%M%S.%f')
+                      if sk.start_time is not None
+                      else ''))
```

Comment:
```haskell
fmap .strftime sk.start_time
```
... would be nice ;)

But jokes aside, how about DRYing this up so any changes can be kept in one place?

```
def strftime_or(thing, default):
  if not thing:
     default
  else:
     thing.strftime("THE FORMAT")
```